app/helpers/fhir_helper.rb

```diff
@@ -374,7 +384,15 @@ def from_us_core_birthsex(patient)
 
   # Given a language string, try to find the corresponding BCP 47 code for it and construct a FHIR::Coding.
   def language_coding(language)
-    PatientHelper.languages(language&.downcase) ? FHIR::Coding.new(**PatientHelper.languages(language&.downcase)) : nil
+    mapped_lang = normalize_and_get_language_name(language)
+    return language if mapped_lang.nil? # Patients should not have invalid languages, but still safer to check here
+
+    language = all_languages[mapped_lang.to_sym]
+    fhir_coding = FHIR::Coding.new
+    fhir_coding.code = language[:iso6391code]
```

Comment:
I think here this line should be:
```ruby
    fhir_coding.code = language[:iso6391code] || mapped_lang
```
That way if the language has an `iso6391code`, we use that, but we fall back on the three letter code if need be. This aligns with the standard, prefer the 2 letter code if it exists, but if not go 3 letters.

Reply:
Good point! Fixing now...